File: backend/src/main/java/com/apocalypse/thefall/controller/character/stats/SpecialController.java
package com.apocalypse.thefall.controller.character.stats;

import com.apocalypse.thefall.dto.character.stats.SpecialDto;
import com.apocalypse.thefall.mapper.character.stats.SpecialMapper;
import com.apocalypse.thefall.service.character.stats.SpecialService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/specials")
@RequiredArgsConstructor
public class SpecialController {

    private final SpecialService specialService;
    private final SpecialMapper specialMapper;

    @GetMapping("/all")
    public List<SpecialDto> getAll() {
        return specialMapper.toDto(specialService.findAll());
    }
}

File: backend/src/main/java/com/apocalypse/thefall/controller/character/stats/SpecialController.kt
package com.apocalypse.thefall.controller.character.stats

import com.apocalypse.thefall.dto.character.stats.DataItemDto
import com.apocalypse.thefall.mapper.character.stats.special.toListDataItemDto
import com.apocalypse.thefall.service.character.stats.SpecialService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/specials")
class SpecialController(
    private val specialService: SpecialService
) {

    @GetMapping("/all")
    fun getAll(): List<DataItemDto> = specialService.findAll().toListDataItemDto()
}